packages/core/src/conflict-resolution.ts

```diff
@@ -253,10 +260,11 @@ export class ConflictResolution {
         // it's an anchor commit
         // TODO: Anchor validation should be done by the StreamHandler as part of applying the anchor commit
         await verifyAnchorCommit(this.dispatcher, this.anchorService, commit);
-        state = await handler.applyCommit(commit, cid, this.context, state);
+        state = await handler.applyCommit(commit, { cid: cid }, this.context, state);
       } else {
         // it's a signed commit
-        const tmpState = await handler.applyCommit(commit, cid, this.context, state);
+        const timestamp = StreamUtils.commitTimestamp(log, commit)
```

Comment:
Seems inefficient to add an `O(n)` operation here. We should use the recursive nature of `fetchLog` instead!

Reply:
Yup. Changed. Now fetchLog provides timestamps.